src/SlackGhost.ts

```diff
@@ -147,9 +147,9 @@ export class SlackGhost {
             throw Error('No intent associated with ghost');
         }
         let changed = false;
-        if (slackUser.profile.display_name && this.displayName !== slackUser.profile.display_name) {
-            await this._intent.setDisplayName(slackUser.profile.display_name);
-            this.displayname = slackUser.profile.display_name;
+        if (slackUser.profile.real_name && this.displayName !== slackUser.profile.real_name) {
```

Comment:
I'm a bit confused about this. I would expect the Display Name to be the thing we want to use (when it's set). AFAIK that's what Slack uses as the the "user name". If the user does not have a Display Name, it will fall back to the "Full name" (which I'm assuming is the `profile.real_name`?).

Should we not keep the same behaviour on Matrix side? This would be:

- If Display Name is set, use that
- If Display Name is not set, use Full Name

Reply:
In Slack, majority users have the username set as Display name and some have it set to their full name. Using the display name on matrix side is the sensible thing to do except majority of users don't set it explicitly to their full name, which is why this sane default in bridge code did not work for us.

In Matrix, we just used their full name as their name. This gave us the ability to `@` mention them using both their username and name. Now, with name gone, you can't `@` mention users using their name, which is what we want to restore.

Reply:
If I understood correctly, we want the "user name" on Matrix to be the same as what Slack shows as the "user name". Could you confirm whether this is what we want, or whether I misunderstood? It's not clear to me yet and I would like to be sure of this.

Reply:
Matrix side username is what we have chosen for each Slack user, based on the `user_nicename` of their WP.org account. The name (equivalent to Display name) on Matrix side was chosen as their actual name (full name).

This got changed by team sync code and was updated to whatever is set as Display name on Slack side. So effectively we just want to revert that and restore the ability to `@` mention users both ways, as it was possible earlier.

Reply:
We discussed this in a DM, purpose is now clear. Resolving.

Reply:
We discussed and concluded that we want the real name to be used here.